### ISSUE TITLE
BertEmbeddings unnecessarily moves embeddings to CPU

### ISSUE THREAD

**Issue by shoarora** [flairNLP/flair#1266]
**Describe the bug**
in `_add_embeddings_internal`, we call `.detach()` and `.cpu()` unnecessarily inside the double for-loop [here](https://github.com/zalandoresearch/flair/blob/master/flair/embeddings.py#L2203).  

1. this whole block is run inside `torch.no_grad()`, so `.detach()` is basically a no-op.
2. `token.set_embedding()` should handle moving the tensor to the right device.  the only reason I can think of for moving off of GPU in the loop is that you need to make room to do something else immediately, but I can't see where that would be the case, since this entire block of code is dedicated to just aggregating the correct tensors.

Moving each layer's token embedding onto CPU one at a time is an unnecessary slowdown, when it could be done after all the tensors are concatenated together.  

**To Reproduce**
```python
# in embedding.py
for token_index, _ in enumerate(feature.tokens):
                    all_layers = []
                    for layer_index in self.layer_indexes:
                        if self.use_scalar_mix:
                            layer_output = all_encoder_layers[int(layer_index)][
                                sentence_index
                            ]
                        else:
                            layer_output = (
                                all_encoder_layers[int(layer_index)]
                                .detach()                               # <---- remove this line
                                .cpu()[sentence_index]        # <---- and this cpu() call
                            )
                        all_layers.append(layer_output[token_index])
```
Toggle those two lines of code, and do a timing comparison on `BertEmbeddings.embed()`

**Expected behavior**

I have limited devices/environments I can test on, but from what I've seen, this should give significant speed-up in runtime.  The exact improvement is highly dependent on both batch size and sentence length since it's inside this double for-loop.  

**Screenshots**

**Environment (please complete the following information):**
 - OS [e.g. iOS, Linux]: Ubuntu 16.04
 - Version: flair-0.4.4

run on an NVIDIA T4

**Additional context**

I'll try to put together a gist with the exact profiling code and outputs that I observed, but in the meantime this should be enough to begin the discussion.  Moreover, I could totally just be missing some important part of the system that explains these lines.

EDIT [gist is here](https://gist.github.com/shoarora/a8aa6ac39ee814b667907ced4bfda6c9)
I compared the flair runtime against just the huggingface runtime for the same model.  flair should take longer, since it runs the underlying huggingface model and does additional processing, but the `flair_runtime/huggingface_runtime` ratio decreases after removing the two lines mentioned
